qualtran/bloqs/chemistry/quad_fermion/givens_bloq.py

```diff
@@ -0,0 +1,218 @@
+            target_i=QBit(),
+            target_j=QBit(),
+            rom_data=QFxp(self.phasegrad_bitsize, self.phasegrad_bitsize, signed=False),
+            phase_gradient=QFxp(self.phasegrad_bitsize, self.phasegrad_bitsize, signed=False),
+        )
+
+    def build_composite_bloq(
+        self,
+        bb: BloqBuilder,
+        target_i: SoquetT,
+        target_j: SoquetT,
+        rom_data: SoquetT,
+        phase_gradient: SoquetT,
+    ) -> Dict[str, SoquetT]:
+        # set up rz-rotation via phase-gradient state
+        add_into_phasegrad_gate = AddIntoPhaseGrad(
+            x_bitsize=self.phasegrad_bitsize,
+            phase_bitsize=self.phasegrad_bitsize,
+            right_shift=0,
+            sign=1,
+            controlled=1,
+        )
+
+        # clifford block
+        target_i = bb.add(XGate(), q=target_i)
+        target_j = bb.add(SGate(), q=target_j)
+        target_j, target_i = bb.add(CNOT(), ctrl=target_j, target=target_i)
+        target_j = bb.add(Hadamard(), q=target_j)
+        target_i = bb.add(SGate(is_adjoint=True), q=target_i)
+        target_j, target_i = bb.add(CNOT(), ctrl=target_j, target=target_i)
+
+        # parallel rz (Can probably be improved with single out of place adder into a single ancilla
+        target_i, rom_data, phase_gradient = bb.add(
+            add_into_phasegrad_gate, x=rom_data, phase_grad=phase_gradient, ctrl=target_i
+        )
+        target_j, rom_data, phase_gradient = bb.add(
+            add_into_phasegrad_gate, x=rom_data, phase_grad=phase_gradient, ctrl=target_j
+        )
+
+        # clifford block
+        target_j, target_i = bb.add(CNOT(), ctrl=target_j, target=target_i)
+        target_i = bb.add(XGate(), q=target_i)
+        target_i, target_j = bb.add(CNOT(), ctrl=target_i, target=target_j)
+        target_j = bb.add(Hadamard(), q=target_j)
+        target_j, target_i = bb.add(CNOT(), ctrl=target_j, target=target_i)
+        target_i = bb.add(SGate(), q=target_i)
+
+        return {
+            'target_i': target_i,
+            'target_j': target_j,
+            'rom_data': rom_data,
+            'phase_gradient': phase_gradient,
+        }
+
+    # def _t_complexit_(self)-> 'TComplexity':
+    #     return TComplexity(clifford=12)
+
+
+@frozen
+class ComplexGivensRotationByPhaseGradient(Bloq):
+    r"""Complex Givens rotation corresponding to a 2-fermion mode transformation generated by
+
+    $$
+        e^{i \phi n_{j}}e^{\theta (a_{i}^{\dagger}a_{j} - a_{j}^{\dagger}a_{i})} =  e^{i \phi Z_{j}/2}e^{i \theta (YX + XY) / 2}
+    $$
+
+    corresponding to the circuit
+
+        i: ───X───X───S^-1───X───Rz(theta)───X───X───@───────X──S^-1─────
+                  │          │               │       │       │
+        j: ───S───@───H──────@───Rz(theta)───@───────X───H───@──Rz(phi)──
+
+    The rotation is performed by addition into a phase state and the fractional binary for
+    $\theta$ is stored in an additional register.
+
+    Args:
```

Comment:
Ditto the comments above about adding a Registers block to your docstring to docusment the signature.